pkg/store/witness/witness.go

```diff
@@ -306,3 +310,45 @@ func (s *Store) AddProof(anchorID, witness string, p []byte) error { //nolint:fu
 
 	return nil
 }
+
+// HandleExpiredKeys is expired keys inspector/handler.
+func (s *Store) HandleExpiredKeys(keys ...string) error {
+	if len(keys) == 0 {
+		logger.Debugf("no expired keys found - all anchors have been processed")
+
+		return nil
+	}
+
+	uniqueAnchors := make(map[string]bool)
+
+	for _, key := range keys {
+		tags, err := s.store.GetTags(key)
+		if err != nil {
+			logger.Errorf("get tags for expired key[%s]: %s", key, err)
+
+			return nil
+		}
+
+		for _, tag := range tags {
+			if tag.Name == anchorIndex {
+				anchor, err := base64.RawURLEncoding.DecodeString(tag.Value)
+				if err != nil {
+					logger.Errorf("failed to decode encoded anchor[%s]: %s", tag.Value, err)
+
+					return nil
+				}
+
+				uniqueAnchors[string(anchor)] = true
+			}
+		}
+	}
+
+	anchors := make([]string, 0, len(uniqueAnchors))
+	for a := range uniqueAnchors {
+		anchors = append(anchors, a)
+	}
+
+	logger.Errorf("failed to process anchors: %s", anchors)
```

Comment:
There's always an error being logged here? Is this right?

Reply:
@DRK3 yes

Reply:
I see, any anchors here represent ones that didn't get processed, so you log here and then when this function returns the expiry service deletes them

Reply:
correct